client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -292,12 +292,7 @@ public void shutdown(final boolean shutdownFactory) {
 
     @Override
     public Set<String> getPublishTopicList() {
-        Set<String> topicList = new HashSet<String>();
-        for (String key : this.topicPublishInfoTable.keySet()) {
-            topicList.add(key);
-        }
-
-        return topicList;
+        return new HashSet<>(this.topicPublishInfoTable.keySet());
```

Comment:
Diamond types are not supported at language level '6'.
It should be like
```java
return new HashSet<String>(this.topicPublishInfoTable.keySet());
```

Reply:
ok , i had add generic type to hashmap for jdk 1.6